### PR TITLE
Cross-compile to Play 2.7

### PR DESCRIPTION
**Pull request by jeffmay**
I was able to add support for Play 2.7 using the same source code by using a few source code compatibility hacks.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 language: scala
 scala:
-  - 2.11.11
+  - 2.12.8
```
Comment (jrouly) [inline .travis.yml:3]: 2.12?

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ branches:
   - master
 
 script:
-  - sbt clean coverage +test coverageReport
+  - sbt clean +coverage +test coverageAggregate
```
Comment (jrouly) [inline .travis.yml:12]: Does `coverageAggregate` need a `+`?

File: code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala [hunk 3]
```diff
@@ -232,6 +234,53 @@ class DocumentedRouter(
...
+    *
+    * If the method exists on the class at compile-time, then it will use that, however, if not, the compiler
+    * will find this implicit class and apply the appropriate adapters.
+    */
+  private implicit class NewRequestHeaderShim(val rh: RequestHeader) extends AnyVal {
+
+    /**
+      * This adds the `.target` method to [[RequestHeader]] that is available in versions of Play 2.6 and above.
+      *
+      * @note this does not return the RequestTarget class, because this class is not available in Play 2.5, but we
```
Comment (macmacbr) [inline code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala:248]: 💯

File: code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala [hunk 4]
```diff
@@ -232,6 +234,53 @@ class DocumentedRouter(
...
+      */
+    def withTarget(target: RequestTargetShim): RequestHeader = {
+      rh.copy(path = target.path)
+    }
+  }
+
+  // This is never used. It only exists to make Play 2.6 and above code compile with the Play 2.5 shims above
+  private implicit class UnusedRequestHeaderShim(rh: RequestHeader) {
+    def copy(path: String): RequestHeader = {
+      throw new UnsupportedOperationException("LegacyRequestHeaderCopyShim.copy should never be called")
```
Comment (macmacbr) [inline code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala:265]: 👍

File: project/build.properties [hunk 5]
```diff
@@ -1 +1 @@
-sbt.version=1.2.3
+sbt.version=1.3.0
```
Comment (kionka-rally) [inline project/build.properties:1]: Why 1.3.0 instead of 1.3.9?

File: .travis.yml (hunk 1 above)
Comment (jducoeur) [reply .travis.yml:3]: I guess the question is, what is this Scala version used for, in this Travis environment?

File: code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala (hunk 4 above)
Comment (jducoeur) [inline code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala:265]: `LegacyRequestHeaderCopyShim` isn't actually the name of this class -- is that intentional?

File: project/Dependencies.scala [hunk 6]
```diff
@@ -4,17 +4,18 @@ object Dependencies {
...
   final val Scala_2_12 = "2.12.6"
+  final val Scala_2_13 = "2.13.1"
 
   final val Play_2_5 = "2.5.18"
   final val Play_2_6 = "2.6.19"
+  final val Play_2_7 = "2.7.4"
 
   private final val guiceVersion = "4.0"
   private final val play25JsonVersion = Play_2_5
   private final val play26JsonVersion = "2.6.9"
```
Comment (jducoeur) [inline project/Dependencies.scala:15]: The `play26JsonVersion` doesn't match the `Play_2_6` version -- is that intentional?  While it isn't the end of the world, it's surprising...

File: project/build.properties (hunk 5 above)
Comment (jeffmay) [reply project/build.properties:1]: There is an issue with SBT `1.3.1` and above and our `sbt-git-versioning` plugin. Once we migrate everything on the Rally side to use sbt 1.x, then we can upgrade our MiMa plugin to the latest version to support upgrading to SBT 1.3.1 or above.

File: code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala (hunk 4 above)
Comment (jeffmay) [reply code/src/test/scala/com/rallyhealth/playmodule/jsonerrors/PlayJsonHttpErrorHandlerSpec.scala:265]: Oops... I renamed the class. I'll update that here.

File: .travis.yml (hunk 2 above)
Comment (jeffmay) [reply .travis.yml:12]: tbh... I don't know

File: .travis.yml (hunk 1 above)
Comment (jeffmay) [reply .travis.yml:3]: This avoids SBT having to download this version of Scala. It saves a little bit of time and avoids the issue I was running into with SBT attempting to download Scala.

File: .travis.yml (hunk 2 above)
Comment (jeffmay) [reply .travis.yml]: I reverted since I don't have a reason to make this change.

File: project/Dependencies.scala (hunk 6 above)
Comment (jeffmay) [reply project/Dependencies.scala:15]: It's because they didn't release a "2.6.9" version of Play, but they did do a selective release for Play Json 🤷‍♂

**Pull request merged**